geopandas/geoseries.py

```diff
@@ -75,6 +76,7 @@ def _geoseries_expanddim(data=None, *args, **kwargs):
     return _expanddim_logic(df)
 
 
+@pandas.api.extensions.register_series_accessor("geo")
```

Comment:
I don't think you should be registering GeoSeries as the accessor, but an accessor class (perhaps derived from GeoSeries so that all the methods exist). This seems a bit counter to the extension registration documentation above. Implemented like this, we don't conform to the documented expectation:
> For consistency with pandas methods, you should raise an AttributeError if the data passed to your accessor has an incorrect dtype.

I think we should be doing this. 

Also, with GeoSeries as the accessor, this would automatically coerce object dtypes, e.g.
```python
s = pd.Series([Point(x, x) for x in range(3)])
x = s.geo.x # this works, even though s is not GeometryDtype
```
I would expect the accessor should only work for GeometryDtype Series.

Reply:
Makes sense. I can look into providing a separate accessor class that delegates to GeoSeries and raises AttributeError for incorrect dtype.

Reply:
Done in 42ea6c5ed89ab8666bb4c5fefbcaa09bc4a5b09b